torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -409,6 +409,24 @@ ProcessGroupNCCL::ProcessGroupNCCL(
         std::string(NCCL_BLOCKING_WAIT));
   }
 
+  // If single-process single-device mode, WorkNCCL::getFuture is supported,
+  // so get a dedicated stream for each device to run FutureNCCL then callbacks.
+  // Depending on the device index of collective outputs, WorkNCCL passes
+  // the corresponding device's then callback stream to FutureNCCL.
+  futureNCCLCallbackStreams_.reserve(c10::cuda::device_count());
+  try {
+    for (int device_index = 0; device_index < c10::cuda::device_count();
+         device_index++) {
+      futureNCCLCallbackStreams_.push_back(
+          std::make_shared<at::cuda::CUDAStream>(
+              at::cuda::getStreamFromPool(device_index)));
+    }
+  } catch (const std::exception& e) {
+    throw std::runtime_error(
+        "Error when inititalizing futureNCCLCallbackStreams: " +
+        std::string(e.what()));
+  }
```

Comment:
Why do we have a try-catch here which basically just rethrows the exception? We can just avoid having a try-catch completely and let the original exception propagate?

Reply:
I was getting unhandled exception when trying the static int that I mentioned on top:
https://github.com/pytorch/pytorch/pull/43447#discussion_r478490040

That's why I put try-catch there.